### ISSUE TITLE
Train prompts for abnormalities

### ISSUE THREAD

**Issue by dpyneo** [johnryh/Face_Embedding_GAN#1]
\Face_Embedding_GAN-master>python -W ignore train.py --GPU 1 --phase 1 --smooth 1 --size 32 --epoch 75 --batch_size 256 --lr 0.001 --n_critic 1 --use_embedding 1
config
loading: [32, 32]
output: [8, 8]

D:\Prog\Face_Embedding_GAN-master>python -W ignore train.py --GPU 1 --phase 1 --smooth 1 --size 32 --epoch 75 --batch_size 256 --lr 0.001 --n_critic 1 --use_embedding 1
config
loading: [32, 32]
output: [8, 8]

Start_jobs_embedding.sh, which is executed separately, is the first one. When tracing debugging, it is found that the file celeba-r32.tfrecords is taken. Shouldn't it be celeba-r8.tfrecords?

Is it because I only have one GPU, and you use four, how can I start training with one GPU? And would you like to know the version of Python and tensorflow you use? The project is very good. Thank you very much.
> face_embedding_gan-master\train.py(37)<module>()
-> real_img, mask = build_input_pipline(batch_size, train_tfrecord_path)
(Pdb) p train_tfrecord_path
'/data/celeba-r32.tfrecords'

**Comment by dpyneo** [comment]
python3 -W ignore train.py --GPU 4 --phase 2 --smooth 1 --size 32 --epoch 40 --batch_size 128 --lr 0.001 --n_critic 1 --use_embedding 1 #16 
It is possible to execute this article alone, but the result is not correct, 0%, and soon quit. Can this not be executed on Win10, using Titan RTX can not be executed, or is it due to insufficient memory and resources?

D:\Prog\Face_Embedding_GAN-master>python -W ignore train.py --GPU 1 --phase 2 --smooth 1 --size 32 --epoch 40 --batch_size 128 --lr 0.001 --n_critic 1 --use_embedding 1
config
loading: [32, 32]
output: [16, 16]
prev_phase_iter: 0
WARNING: Logging before flag parsing goes to stderr.
W0707 15:56:37.759290  9556 deprecation_wrapper.py:119] From train.py:35: The name tf.reset_default_graph is deprecated. Please use tf.compat.v1.reset_default_graph instead.

W0707 15:56:37.805168  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\Input_Pipeline_celeba.py:52: The name tf.parse_single_example is deprecated. Please use tf.io.parse_single_example instead.

W0707 15:56:37.805168  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\Input_Pipeline_celeba.py:52: The name tf.FixedLenFeature is deprecated. Please use tf.io.FixedLenFeature instead.

W0707 15:56:37.819130  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\Input_Pipeline_celeba.py:32: The name tf.random_uniform is deprecated. Please use tf.random.uniform instead.

W0707 15:56:37.838052  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\Input_Pipeline_celeba.py:71: The name tf.image.resize_images is deprecated. Please use tf.image.resize instead.

W0707 15:56:37.966171  9556 deprecation.py:323] From D:\Prog\Face_Embedding_GAN-master\Input_Pipeline_celeba.py:96: DatasetV1.make_one_shot_iterator (from tensorflow.python.data.ops.dataset_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `for ... in dataset:` to iterate over a dataset. If using `tf.estimator`, return the `Dataset` object directly from your input function. As a last resort, you can use `tf.compat.v1.data.make_one_shot_iterator(dataset)`.
W0707 15:56:37.975146  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\network_utility.py:362: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.

Tensor("Reshape:0", shape=(128, 16, 16, 3), dtype=float32) Tensor("Reshape_1:0", shape=(128, 16, 16, 1), dtype=float32)
W0707 15:56:37.979135  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\network_utility.py:378: The name tf.train.AdamOptimizer is deprecated. Please use tf.compat.v1.train.AdamOptimizer instead.

Solver Configured
W0707 15:56:38.363300  9556 lazy_loader.py:50]
The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
  * https://github.com/tensorflow/io (for I/O related ops)
If you depend on functionality not listed there, please file an issue.

W0707 15:56:38.538857  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\network_utility.py:286: The name tf.nn.avg_pool is deprecated. Please use tf.nn.avg_pool2d instead.

W0707 15:56:38.539827  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\network_utility.py:287: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.

var_list initiated
W0707 15:56:38.804149  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\network_utility.py:438: The name tf.summary.scalar is deprecated. Please use tf.compat.v1.summary.scalar instead.

W0707 15:56:38.810136  9556 deprecation_wrapper.py:119] From D:\Prog\Face_Embedding_GAN-master\network_utility.py:445: The name tf.train.Saver is deprecated. Please use tf.compat.v1.train.Saver instead.

W0707 15:56:39.228886  9556 deprecation.py:323] From C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\tensorflow\python\ops\math_grad.py:1205: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
Discriminator Total parameters:16.055357M
Generator Total parameters:18.168158M
Total Total parameters:34.223515M
W0707 15:56:40.330911  9556 deprecation_wrapper.py:119] From train.py:48: The name tf.summary.merge_all is deprecated. Please use tf.compat.v1.summary.merge_all instead.

W0707 15:56:40.886480  9556 deprecation_wrapper.py:119] From train.py:60: The name tf.summary.FileWriter is deprecated. Please use tf.compat.v1.summary.FileWriter instead.

Session Initiated
  0%|                                                                                        | 0/21875 [00:00<?, ?it/s]
D:\Prog\Face_Embedding_GAN-master>

**Comment by dpyneo** [comment]
I would also like to ask if there are any plans to make your model public in the near future.

**Comment by johnryh** [comment]
In the command `--size 32` is corresponding to the file `celeba-r32.tfrecords`. Make sure you have the proper tfrecord downloaded. Or you can just use `--size 8` for `celeba-r8.tfrecords`. Should work the same for phase 1 training.  Please note the input-pipeline will resize the input images to the corresponding phase's output dimension.  Also note if you want to start from phase > 1 without loading the previous phases' weights you need to modify the model.loader logic in `train.py`

**Comment by johnryh** [comment]
Currently, there is no plan for releasing the trained weights in the short term. We are still working on improving the reliability and variation of our model.

**Comment by dpyneo** [comment]
Thank you very much for your reply. I'll try to revise it then. I expect you to succeed in a very good project.
